src/main/java/org/opensearch/agent/tools/SearchIndexTool.java

```diff
@@ -97,15 +97,9 @@ public <T> void run(Map<String, String> parameters, ActionListener<T> listener)
             try {
                 searchRequest = getSearchRequest(index, query);
             } catch (Exception e1) {
-                try {
-                    // try different json parsing method
-                    query = jsonObject.get(QUERY_FIELD).getAsString();
-                    searchRequest = getSearchRequest(index, query);
-                } catch (Exception e2) {
-                    // try wrapped query
-                    query = "{\"query\": " + query + "}";
-                    searchRequest = getSearchRequest(index, query);
-                }
+                // try different json parsing method
+                query = jsonObject.get(QUERY_FIELD).getAsString();
+                searchRequest = getSearchRequest(index, query);
```

Comment:
Remove this since it would cause exception when `jsonObject.get(QUERY_FIELD)` returns jsonElement. And add error log here.